src/Migration/Adapter/Database/MySqlSchemaAdapter.php

```diff
@@ -178,10 +202,18 @@ protected function getTables(): array
                 AND t.table_schema=database()
                 AND t.table_type = 'BASE TABLE'";
 
+        if ($tableNames !== null) {
+            if (empty($tableNames)) {
+                return [];
```

Comment:
What is this for a special case?

Reply:
$tableNames could be null or empty array - this cases are different. When it is null, we just get all tables, so we don't need any conditions in sql. When it's empty array, the sql condition becomes invalid (`in ()`), and even if it was valid, it's always equals to `false`, so we can just return an empty array. Yes, we can change it to returning all tables in case of empty array, but this would be looking strange, imho.

Thanks for comments, I will fix descriptions. I could change the first check to `is_array($tableNames)`, but I would rather leave this part if you have not any suggestions.